### ISSUE TITLE
Basket Access Tier 2 not triggered by REST PUT api

### ISSUE THREAD

**Issue by dsibilio** [juice-shop/juice-shop#803]
When requesting the **/api/BasketItems/{id}** endpoint with **PUT** HTTP method, using the token of user A (linked to basketId=1), to insert a completely new product in the basket of user B (linked to basketId=2), the **Basket Access Tier 2 challenge doesn't get solved**.

Screenshots below:

![image](https://user-images.githubusercontent.com/24280982/52584632-d77c4980-2e32-11e9-840c-a789cb5e0297.png)
_Fig. 1 - A successful request that inserted a new item in basket 2_

![image](https://user-images.githubusercontent.com/24280982/52584693-0abed880-2e33-11e9-92d0-a47dc1c3cd83.png)
_Fig. 2 - Requesting the basket 2 to see its content, the item is inserted once (wasn't present before)_

![image](https://user-images.githubusercontent.com/24280982/52584025-540e2880-2e31-11e9-9d0b-ec03f9201476.png)
_Fig. 3 - The visual representation of user B's basket._

![image](https://user-images.githubusercontent.com/24280982/52584137-96d00080-2e31-11e9-9e22-4ee7f83abb19.png)
_Fig. 4 - The challenge has not been completed :-(_

**Comment by bkimminich** [comment]
Hmm, adding new items should only work with POST, so it only checks POST requests for challenge condition. Seems you found a nice additional path to victory! Thanks for reporting!

**Comment by dsibilio** [comment]
Yeah, I guess it's actually updating a pre-existing item from any basket and linking it to the desired basket.
It is de facto possible to add new items in other users' baskets  this way, and should most likely count as a valid solution for this challenge 👍 

Thanks for the quick reply and thank you for your awesome work!

**Comment by bkimminich** [comment]
Ok, so you essentially move the product from some user's basket into basket of user B! I might even make that a challenge of its own!

**Comment by bkimminich** [comment]
With the next release your `PUT` path should not work any more thanks to the nice `node-sequelize-noupdate-attributes` plugin. If you find it still works on https://juice-shop-staging.herokuapp.com (after https://travis-ci.org/bkimminich/juice-shop/builds/492128418 finishes successfully) please reopen this ticket. Thanks!

**Comment by JRKelso** [comment]
I'm running version 8.5.1 and retried a similar issue I found back in January (Sorry, forgot to report) and find it still works. The steps to reproduce are as follows:

Find a list of baskets with items (GET /rest/basket/<id>)
Insert an un-basketed item (POST /api/BasketItems/ {"quantity":1, "ProductId":1} )
Modify that BasketItem with the BasketId (PUT /api/BasketItems/<id> {"BasketId":<number>})

I tested both staging and my local 8.5.1 and find it works. Probably because I'm not changing from one number to another, because if you try to update that now basket-ed item, it throws `{"message":"internal error","errors":["sequelize.ValidationErrorItem is not a constructor"]}`. Perhaps making it so BasketItems can't have nullable BasketIds?
